src/test/java/com/puppycrawl/tools/checkstyle/CheckerTest.java

```diff
@@ -743,6 +779,28 @@ protected void processFiltered(File file, List<String> lines) throws CheckstyleE
         }
     }
 
+    private static class DummyFileSetViolationCheck extends AbstractFileSetCheck
+        implements ExternalResourceHolder {
```

Comment:
I changed this to a normal FileSet with just a `log` in `processFiltered` and no `finishProcessing` or `fireErrors`.
Is this ok? Or do you still want `AbstractCheck`?